File: replace_icons.js
const fs = require('fs');
const path = require('path');
const https = require('https');

const dirs = [
  path.join(__dirname, 'kqDJT8D'),
  path.join(__dirname, 'yoTDTIS')
];

let filesToProcess = [];

function findFiles(d) {
  const files = fs.readdirSync(d);
  for (let f of files) {
    const full = path.join(d, f);
    if (fs.statSync(full).isDirectory()) findFiles(full);
    else if (f.endsWith('.html') || f.endsWith('.js')) {
      filesToProcess.push(full);
    }
  }
}
dirs.forEach(findFiles);

const uniqueIcons = new Set();
const fileIcons = {};

const re = /<i\s+[^>]*data-lucide=["']([^"']+)["'][^>]*><\/i>/gi;

filesToProcess.forEach(file => {
  const content = fs.readFileSync(file, 'utf8');
  let match;
  fileIcons[file] = [];
  while ((match = re.exec(content)) !== null) {
    uniqueIcons.add(match[1]);
    fileIcons[file].push({
      fullMatch: match[0],
      iconName: match[1]
    });
  }
});

console.log('Unique icons:', Array.from(uniqueIcons));

// Download all SVGs
async function fetchSvg(name) {
  return new Promise((resolve, reject) => {
    https.get(`https://unpkg.com/lucide-static@0.344.0/icons/${name}.svg`, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        if (res.statusCode === 200) resolve(data);
        else resolve(null);
      });
    }).on('error', reject);
  });
}

async function run() {
  const svgMap = {};
  for (const icon of uniqueIcons) {
    console.log('Fetching', icon);
    const svg = await fetchSvg(icon);
    if (svg) {
      svgMap[icon] = svg;
    } else {
      console.log('FAILED to fetch', icon);
    }
  }

  // Replace in files
  for (const file of filesToProcess) {
    const matches = fileIcons[file];
    if (!matches || matches.length === 0) continue;
    
    let content = fs.readFileSync(file, 'utf8');
    let changed = false;
    
    const replaceRe = /<i\s+(.*?)(data-lucide=["']([^"']+)["'])(.*?)><\/i>/gi;
    
    content = content.replace(replaceRe, (match, before, dataAttr, iconName, after) => {
      const svg = svgMap[iconName];
      if (!svg) return match;
      
      let cls = '';
      const clsMatch = match.match(/class=["']([^"']+)["']/i);
      if (clsMatch) {
        cls = clsMatch[1];
      }
      
      let newSvg = svg.replace(/class="[^"]*"/, ''); 
      newSvg = newSvg.replace('<svg ', `<svg class="${cls}" `);
      changed = true;
      return newSvg;
    });
    
    if (changed) {
      fs.writeFileSync(file, content, 'utf8');
      console.log('Updated', file);
    }
  }
}

run();

File: replace_icons.py
import os
import re
import urllib.request
import json

dirs = [
    os.path.join(os.path.dirname(__file__), 'kqDJT8D'),
    os.path.join(os.path.dirname(__file__), 'yoTDTIS')
]

files_to_process = []
for d in dirs:
    for root, _, files in os.walk(d):
        for f in files:
            if f.endswith('.html') or f.endswith('.js'):
                files_to_process.append(os.path.join(root, f))

unique_icons = set()
file_icons = {}

# Match <i ... data-lucide="icon-name" ...></i>
icon_re = re.compile(r'<i\s+[^>]*data-lucide=["\']([^"\']+)["\'][^>]*></i>', re.IGNORECASE)

for file in files_to_process:
    try:
        with open(file, 'r', encoding='utf-8') as f:
            content = f.read()
    except Exception as e:
        continue

    matches = icon_re.finditer(content)
    file_icons[file] = []
    for match in matches:
        icon_name = match.group(1)
        unique_icons.add(icon_name)
        file_icons[file].append(icon_name)

print(f"Unique icons to fetch: {unique_icons}")

svg_map = {}
for icon in unique_icons:
    url = f"https://unpkg.com/lucide-static@0.344.0/icons/{icon}.svg"
    print(f"Fetching {icon}...")
    try:
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
        with urllib.request.urlopen(req) as response:
            if response.status == 200:
                svg_map[icon] = response.read().decode('utf-8')
            else:
                print(f"Failed to fetch {icon}")
    except Exception as e:
        print(f"Error fetching {icon}: {e}")

def main():
    for file in files_to_process:
        if not file_icons.get(file):
            continue

        try:
            with open(file, 'r', encoding='utf-8') as f:
                content = f.read()
        except Exception:
            continue

        changed = False

        def replace_func(match):
            nonlocal changed
            full_match = match.group(0)
            icon_name = match.group(2)
            
            svg = svg_map.get(icon_name)
            if not svg:
                return full_match

            cls_match = re.search(r'class=["\']([^"\']+)["\']', full_match, re.IGNORECASE)
            cls = cls_match.group(1) if cls_match else ""

            new_svg = re.sub(r'class="[^"]*"', '', svg)
            new_svg = new_svg.replace('<svg ', f'<svg class="{cls}" ')
            
            changed = True
            return new_svg

        replace_re = re.compile(r'<i\s+(.*?)(?:data-lucide=["\']([^"\']+)["\'])(.*?)></i>', re.IGNORECASE)
        new_content = replace_re.sub(replace_func, content)

        if changed:
            with open(file, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated {file}")

main()

print("Done.")
